ledger/evalIndexer.go

```diff
@@ -213,5 +213,11 @@ func EvalForIndexer(il indexerLedgerForEval, block *bookkeeping.Block, proto con
 			fmt.Errorf("EvalForIndexer() err: %w", err)
 	}
 
+	err = eval.finalValidation()
```

Comment:
Indexer shouldn't do validation, and this breaks indexer at https://github.com/algorand/go-algorand/pull/2922/files#diff-ad138066c4b3cef29e960244350ba0f9adae9051c61d09162dc511bdaeda7fddR321 since indexer doesn't store any totals. Why was this added?